clang/tools/clang-offload-wrapper/ClangOffloadWrapper.cpp

```diff
@@ -260,6 +271,33 @@ static cl::opt<bool> AddOpenMPOffloadNotes(
 
 namespace {
 
+auto getValueAsStringRef(const Module *M, Constant *Name) {
+  auto Var = M->getGlobalVariable(Name->getName(), true);
+  auto Initializer = Var->getInitializer();
+  if (Initializer->isNullValue()) {
+    // Return data even if zeroinitializer is in IR.
+    // Length of SizeTy is necessary because PropertyValue with Data
+    // puts a bitsize variable of type SizeTy at the start of Data.
+    static char zi[sizeof(llvm::util::PropertyValue::SizeTy)] = {0};
+    static StringRef ZeroInitializer{zi,
+                                     sizeof(llvm::util::PropertyValue::SizeTy)};
+
+    return ZeroInitializer;
+  }
+  auto StringRef = cast<ConstantDataArray>(Initializer)->getRawDataValues();
+  return StringRef;
+}
+
+StringRef removeNullTerminator(StringRef S) {
+  return S.substr(0, S.size() - 1);
+}
+
+auto getInitializerNumElements(Constant *Initializer) {
+  Type *Ty = Initializer->getType();
+  auto *ArrTy = dyn_cast<ArrayType>(Ty);
```

Comment:
What is expected behavior if it's not array type?

Reply:
Initializer must be of ArrayType.  Assertion added.